File: 프로그래머스/lv2/12924. 숫자의 표현/숫자의 표현.py
def solution(n):
    answer, cnt = 0, 1
    while cnt <= n:
        hap = 0
        for i in range(cnt,n+1):
            hap += i
            if hap < n: continue
            else: 
                if hap == n: answer += 1
                cnt += 1
                break
    return answer
solution(15)

File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.java
class Solution {
    public int solution(int n) {
        int answer = 0;
        for (int i=1; i<=n; i++) {
            int num = 0;
            for (int j=i; j<=n; j++) {
                if (num+j>=n) {
                    if (num+j == n) answer++;
                    break;
                }
                else num+=j;
            }
        }
        return answer;
    }
}